File: src/main/java/org/jetbrains/plugins/template/listeners/MyDynamicPluginListener.java
package org.jetbrains.plugins.template.listeners;

import com.intellij.ide.plugins.CannotUnloadPluginException;
import com.intellij.ide.plugins.DynamicPluginListener;
import com.intellij.ide.plugins.IdeaPluginDescriptor;
import com.intellij.openapi.components.ServiceManager;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.plugins.template.services.MyApplicationService;

class MyDynamicPluginListener implements DynamicPluginListener {
    @Override
    public void beforePluginLoaded(@NotNull IdeaPluginDescriptor pluginDescriptor) {
        ServiceManager.getService(MyApplicationService.class);
    }

    @Override
    public void pluginLoaded(@NotNull IdeaPluginDescriptor pluginDescriptor) {
        System.out.println("xx");
    }

    @Override
    public void beforePluginUnload(@NotNull IdeaPluginDescriptor pluginDescriptor, boolean isUpdate) {

    }

    @Override
    public void checkUnloadPlugin(@NotNull IdeaPluginDescriptor pluginDescriptor) throws CannotUnloadPluginException {

    }

    @Override
    public void pluginUnloaded(@NotNull IdeaPluginDescriptor pluginDescriptor, boolean isUpdate) {

    }
}

File: src/main/kotlin/org/jetbrains/plugins/template/listeners/MyDynamicPluginListener.kt
package org.jetbrains.plugins.template.listeners

import com.intellij.ide.plugins.DynamicPluginListener
import com.intellij.ide.plugins.IdeaPluginDescriptor
import com.intellij.openapi.components.ServiceManager
import org.jetbrains.plugins.template.services.MyApplicationService

internal class MyDynamicPluginListener : DynamicPluginListener {

    override fun beforePluginLoaded(pluginDescriptor: IdeaPluginDescriptor) {
        ServiceManager.getService(MyApplicationService::class.java)
    }

    override fun pluginLoaded(pluginDescriptor: IdeaPluginDescriptor) = Unit

    override fun beforePluginUnload(pluginDescriptor: IdeaPluginDescriptor, isUpdate: Boolean) = Unit

    override fun checkUnloadPlugin(pluginDescriptor: IdeaPluginDescriptor) = Unit

    override fun pluginUnloaded(pluginDescriptor: IdeaPluginDescriptor, isUpdate: Boolean) = Unit
}
